### PR TITLE
set fileHandleLimit to 1024

### PR DESCRIPTION
**Pull request by RSulzmann**

### CODE REVIEW

**Comment by markusthoemmes** [comment]
Can you provide a testcase?

**Comment by domdom82** [comment]
@RSulzmann can you shove the DockerOpenFileLimitTest.scala to open? also it needs to increase its limit.

**Comment by RSulzmann** [comment]
Here is an action code (nodejs) to check the file handle limit boundary:

```node
var fs = require('fs');

function main(params) {
	console.log (process.versions)	

    return new Promise((resolve, reject) => {

        fs.writeFile("DATA", "LOTS OF FILES!!!!", function(err) {
        	console.log ("write") 
        	
            if (err) {
            	console.log ("reject write")
                reject(err);
            }

            var latch = params.N;

            function read(iter) {     	
                fs.readFile("DATA", "utf8", function(err, contents) {
                    if (err) {
                    	console.log ("reject read " + iter + " " + err)
                        reject(err);
                    } else if (--latch === 0) {
                    	console.log ("resolve")
                        resolve({N: params.N});
                    }
                });
            }

            for (let i = 0; i < params.N; i++) {
                console.log ("read "+i)
                read(i);
            }
        }); 
    });
}
```

create an action with the code above copied to a file:
**wsk action create _actionName actionFileName_**
invoke with the parameter N 1024:
**wsk action invoke _actionName_ -p N 1024**
check log output:
**wsk activation logs _activationID_**

**Comment by rabbah** [comment]
turn that into a test? (didn't we have one already?)

**Comment by RSulzmann** [comment]
Yes, we have one, that we will move to openwhisk.

**Comment by rabbah** [comment]
@RSulzmann @markusthoemmes is this ready? is there a pg?

**Comment by rabbah** [comment]
PG1/1110.

**Comment by rabbah** [comment]
@markusthoemmes I pg'ed this PG1/1110.

**Comment by rabbah** [comment]
I rebased this pr in my branch and ran new PG which failed. PG1/1124 for reference.

**Comment by rabbah** [comment]
Rebased with companion, pg2/1010 running.

File: tests/src/whisk/core/limits/ActionLimitsTests.scala [hunk 1]
```diff
@@ -150,4 +153,40 @@ class ActionLimitsTests extends TestHelpers with WskTestHelpers {
...
+        (wp, assetHelper) =>
+            val name = "TestFileLimitGood-" + System.currentTimeMillis()
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(openFileAction))
+            }
+
+            val run = wsk.action.invoke(name, Map("numFiles" -> (openFileLimit / 2).toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.status shouldBe "success"
```
Comment (rabbah) [inline tests/src/whisk/core/limits/ActionLimitsTests.scala]: this should be tighter: it should confirm the expected result `{ filesToOpen: numFiles, filesOpen: openFiles.length}`.

File: tests/src/whisk/core/limits/ActionLimitsTests.scala [hunk 2]
```diff
@@ -150,4 +153,40 @@ class ActionLimitsTests extends TestHelpers with WskTestHelpers {
...
+        (wp, assetHelper) =>
+            val name = "TestFileLimitBad-" + System.currentTimeMillis()
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(openFileAction))
+            }
+
+            val run = wsk.action.invoke(name, Map("numFiles" -> (openFileLimit * 2).toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.status should not be "success"
```
Comment (rabbah) [inline tests/src/whisk/core/limits/ActionLimitsTests.scala]: similarly, `activation.response.success shouldBe false`.

**Comment by rabbah** [comment]
Made the fixes suggested here in new PR.

**Pull request closed without merging**